src/bun.js/webcore/streams.zig

```diff
@@ -3721,6 +3744,33 @@ pub const FileSink = struct {
         return .{ .result = {} };
     }
 
+    pub fn onAutoFlush(this: *FileSink) bool {
+        if (this.done or !this.writer.hasPendingData()) {
+            this.updateRef(false);
+            this.auto_flusher.registered = false;
+            return false;
+        }
+
+        const amt = switch (this.writer.flush()) {
+            .done, .wrote, .pending => |amt| amt,
+            else => 0,
+        };
+
+        if (amt == 0) {
+            this.updateRef(false);
+        } else {
+            this.runPending();
```

Comment:
We want to run pending when:
- `done`
- If we wrote part of the data

Instead, we should runPending when `!this.writer.hasPendingData()`

If this leads to a timeout in tests, it's likely due to not draining microtasks since this is being called after the microtask queue is over. If the developer immediately calls write every time the write Promise fulfills, that could maybe lead to an infinite loop but we will have to see.

I think both of these branches `if (amt == 0)` and this one should be removed. And instead